[CI] MixedClusterClientYamlTestSuiteIT class failing

**Build Scans:**
- [elasticsearch-intake #10091 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/2e666h3l5vf4y)
- [elasticsearch-pull-request #32652 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/khqggzj3iokcy)
- [elasticsearch-pull-request #32649 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/piksud4ieci2o)
- [elasticsearch-pull-request #32648 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/7cmvi7yjjwfag)
- [elasticsearch-pull-request #32653 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/qdwqk427o77ho)
- [elasticsearch-pull-request #32646 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/w47zvhdy32mz4)
- [elasticsearch-pull-request #32647 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/znqdecseezzzq)
- [elasticsearch-pull-request #32644 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/52ah4u5pucl6m)
- [elasticsearch-pull-request #32641 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/n63pxctqc3vty)
- [elasticsearch-pull-request #32636 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/r2j7p7mnyrw6u)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.16.0#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search/40_indices_boost/Indices boost with alias}" -Dtests.seed=DD0A7F60139F7BB0 -Dtests.bwc=true -Dtests.locale=sah-Cyrl-RU -Dtests.timezone=America/Santarem -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [search/40_indices_boost:2]: expected [2xx] status code but api [indices.create] returned [503 Service Unavailable] [{"error":{"root_cause":[{"type":"master_not_discovered_exception","reason":null,"stack_trace":"org.elasticsearch.discovery.MasterNotDiscoveredException\n"}],"type":"master_not_discovered_exception","reason":null,"stack_trace":"org.elasticsearch.discovery.MasterNotDiscoveredException\n"},"status":503}]
```

**Issue Reasons:**
- [main] 8 consecutive failures in class org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
- [main] 8 consecutive failures in step 8.16.0_bwc-snapshots
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 7 consecutive failures in pipeline elasticsearch-pull-request
- [main] 99 failures in class org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT (14.1% fail rate in 700 executions)
- [main] 94 failures in step 8.16.0_bwc-snapshots (15.1% fail rate in 623 executions)
- [main] 2 failures in step openjdk21_8.16.0_java-matrix-bwc (10.5% fail rate in 19 executions)
- [main] 3 failures in step 8.16.0_bwc (7.7% fail rate in 39 executions)
- [main] 9 failures in pipeline elasticsearch-intake (4.7% fail rate in 193 executions)
- [main] 83 failures in pipeline elasticsearch-pull-request (18.7% fail rate in 443 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (9.5% fail rate in 21 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

```[2024-09-24T13:27:32,771][ERROR][o.e.ExceptionsHelper     ] [v8.16.0-0] fatal error java.lang.AssertionError: Unknown NamedWriteable [org.elasticsearch.search.DocValueFormat][_ts_routing_hash]
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.ExceptionsHelper.lambda$maybeDieOnAnotherThread$4(ExceptionsHelper.java:445)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at org.elasticsearch.server@9.0.0-SNAPSHOT/org.elasticsearch.ExceptionsHelper.maybeDieOnAnotherThread(ExceptionsHelper.java:435)
	at org.elasticsearch.transport.netty4@9.0.0-SNAPSHOT/org.elasticsearch.transport.netty4.Netty4MessageInboundHandler.exceptionCaught(Netty4MessageInboundHandler.java:67)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:346)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:447)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.109.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.transport@4.1.109.Final/io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.transport@4.1.109.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.109.Final/io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.transport@4.1.109.Final/io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.transport@4.1.109.Final/io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.transport@4.1.109.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689)
	at io.netty.transport@4.1.109.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652)
	at io.netty.transport@4.1.109.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.common@4.1.109.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.common@4.1.109.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.lang.Thread.run(Thread.java:1570)
```

The mixed cluster crashed due to this. Pinging appropriate team.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -313,8 +313,6 @@ tests:
 - class: org.elasticsearch.xpack.inference.InferenceCrudIT
   method: testSupportedStream
   issue: https://github.com/elastic/elasticsearch/issues/113430
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  issue: https://github.com/elastic/elasticsearch/issues/113466
 - class: org.elasticsearch.xpack.ml.integration.MlJobIT
   method: testOutOfOrderData
   issue: https://github.com/elastic/elasticsearch/issues/113477

